cfgmgr/vrfmgr.cpp

```diff
@@ -0,0 +1,146 @@
+#include <string.h>
+#include "logger.h"
+#include "dbconnector.h"
+#include "producerstatetable.h"
+#include "tokenize.h"
+#include "ipprefix.h"
+#include "vrfmgr.h"
+#include "exec.h"
+#include "shellcmd.h"
+
+#define VRF_TABLE_START 100
+#define VRF_TABLE_END 164
+
+using namespace swss;
+
+VrfMgr::VrfMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, const vector<string> &tableNames) :
+        Orch(cfgDb, tableNames)
+{
+    for (uint32_t i = VRF_TABLE_START; i < VRF_TABLE_END; i++)
+    {
+        m_freeTables.emplace(i);
+    }
+}
+
+uint32_t VrfMgr::getFreeTable(void)
+{
+    SWSS_LOG_ENTER();
+
+    if (m_freeTables.empty())
+    {
+        return 0;
+    }
+
+    uint32_t table = *m_freeTables.begin();
+    m_freeTables.erase(table);
+
+    return table;
+}
+
+void VrfMgr::returnTable(uint32_t table)
+{
+    SWSS_LOG_ENTER();
+
+    m_freeTables.emplace(table);
+}
+
+bool VrfMgr::delLink(const string& vrfName)
+{
+    SWSS_LOG_ENTER();
+
+    stringstream cmd;
+    string res;
+
+    if (m_vrfTableMap.find(vrfName) == m_vrfTableMap.end())
+    {
+        return false;
+    }
+
+    returnTable(m_vrfTableMap[vrfName]);
```

Comment:
i feel we should move this after line 69 since we should only recycle the id when it is free.

Reply:
Done